Samples/DirectML_ESRGAN/image_helpers.h

```diff
@@ -0,0 +1,315 @@
+#include <wincodec.h>
+#include <wrl/client.h>
+#include <wil/result.h>
+#include <span>
+#include "half.hpp"
+
+enum class ChannelOrder
+{
+    RGB,
+    BGR,
+};
+
+// Converts a pixel buffer to an NCHW tensor (batch size 1).
+// Source: buffer of RGB pixels (HWC) using uint8 components.
+// Target: buffer of RGB planes (CHW) using float32/float16 components.
+template <typename T> 
+void CopyPixelsToTensor(
+    std::span<const std::byte> src, 
+    std::span<std::byte> dst,
+    uint32_t height,
+    uint32_t width,
+    uint32_t channels)
+{
+    std::span<T> dstT(reinterpret_cast<T*>(dst.data()), dst.size_bytes() / sizeof(T));
+
+    for (size_t pixelIndex = 0; pixelIndex < height * width; pixelIndex++)
+    {
+        float r = static_cast<float>(src[pixelIndex * channels + 0]) / 255.0f;
+        float g = static_cast<float>(src[pixelIndex * channels + 1]) / 255.0f;
+        float b = static_cast<float>(src[pixelIndex * channels + 2]) / 255.0f;
+
+        dstT[pixelIndex + 0 * height * width] = r;
```

Comment:
do you need a static cast depending on the type for T

Reply:
Half has an implicit cast from float16.